src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -225,14 +231,51 @@ internal override void GetDisplayData(VSTREEDISPLAYDATA[] pData)
             pData[0].Image = pData[0].SelectedImage = (ushort)StandardGlyphGroup.GlyphLibrary;
         }
 
-        public static IHierarchicalDifferenceCollection ComputeDiffSpans(SourceText oldText, SourceText newText, ITextDifferencingService diffService, CancellationToken cancellationToken)
+        private static IHierarchicalDifferenceCollection ComputeDiffSpans(ITextDifferencingService diffService, TextDocument left, TextDocument right, CancellationToken cancellationToken)
         {
-            var diffResult = diffService.DiffStrings(oldText.ToString(), newText.ToString(), new StringDifferenceOptions()
+            // TODO: it would be nice to have a syntax based differ for presentation here, 
+            //       current way of just using text differ has its own issue, and using syntax differ in compiler that are for incremental parser
+            //       has its own drawbacks.
```

Comment:
we don't have syntax diffing to use for presentation. only syntax diffing we have is for incremental parser, and that one doesn't give good diff result for presentation.

ex), when there is change from identifier M1 to M2. that diff only gives "1" -> "2" as change. not "M1" -> "M2" since for incremental parsing, it only care about actual change.

also, compiler syntax differ has threshold on how many change it would track, if there are a lot of changes, it will just put whole changes in one chunk and tell all of them as changed since again, that is better for incremental parsing. like I said in the comment, we need better syntax differ for presentation.